### ISSUE TITLE
Token burn on a nft fails

### ISSUE THREAD

**Issue by sourcelib-stack** [bloxbean/cardano-client-lib#28]
Hello,

Trying to execute a token burn on a nft fails with the following error:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: DeserializeError { location: Some("Transaction.body.TransactionBody.outputs.TransactionOutputs.TransactionOutput.amount.Value.multiasset.MultiAsset.Assets.BigNum"), failure: CBOR(Expected(UnsignedInteger, Special)) }', src\transaction.rs:14:61
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: DeserializeError { location: Some("Transaction"), failure: CBOR(NotEnough(0, 0)) }', src\transaction.rs:55:61
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: DeserializeError { location: Some("Transaction"), failure: CBOR(NotEnough(0, 0)) }', src\transaction.rs:55:61
```

The code used to perform the action is the following:

```
MultiAsset multiAsset = new MultiAsset();
        multiAsset.setPolicyId(policyID);
        Asset asset = new Asset("test name", BigInteger.valueOf(-1));
        multiAsset.getAssets().add(asset);

        CBORMetadataMap metadataMapAssetFile = new CBORMetadataMap()
                .put("mediaType", assetMetadata.getFiles().stream().findFirst().get().mediaType)
                .put("name", assetMetadata.getFiles().stream().findFirst().get().name)
                .put("src", assetMetadata.getFiles().stream().findFirst().get().src);

        CBORMetadataList metadataMapAssetFiles= new CBORMetadataList();
        metadataMapAssetFiles.add(metadataMapAssetFile);

        CBORMetadataMap metadataMapAssetDetails = new CBORMetadataMap()
                .put("name", "test name");

        CBORMetadataMap metadataAssetRoot = new CBORMetadataMap()
                .put(assetMetadata.getName(), metadataMapAssetDetails);

        CBORMetadataMap tokenInfoMap = new CBORMetadataMap()
                .put(policyID, metadataAssetRoot);

        Metadata metadata = new CBORMetadata()
                .put(new BigInteger(NFT_JSON_META), tokenInfoMap);



        MintTransaction mintTransaction =
                MintTransaction.builder()
                        .sender(account)
                        .mintAssets(Arrays.asList(multiAsset))
                        .policyScript(scriptPubKey)
                        .policyKeys(sKeys)
                        .build();

        long ttl = backendService.getBlockService().getLastestBlock().getValue().getSlot() + 1000;
        TransactionDetailsParams detailsParams = TransactionDetailsParams.builder().ttl(ttl).build();

        BigInteger fee = backendService.getFeeCalculationService().calculateFee(mintTransaction, detailsParams, metadata);

        mintTransaction.setFee(fee);

        System.out.println("Fees : "+fee);

        Result<TransactionResult> result= backendService.getTransactionHelperService().mintToken(mintTransaction, detailsParams, metadata);
        if(result.isSuccessful())
            System.out.println("Transaction Id: " + result.getValue().getTransactionId());
        else
            System.out.println("Transaction failed: " + result);

```

Thank you very much in advance for your support!

**Comment by satran004** [comment]
Fixed in 0.1.5-beta.

Token burn can be done through low level transaction serialization api. Currently, the high level api "MintTransaction" is for token mint only.

You can check the token burn example here 
https://github.com/bloxbean/cardano-client-examples/blob/main/src/main/java/com/bloxbean/cardano/client/examples/MintAndBurn.java

**Comment by somereason** [comment]
> Fixed in 0.1.5-beta.
> 
> Token burn can be done through low level transaction serialization api. Currently, the high level api "MintTransaction" is for token mint only.
> 
> You can check the token burn example here https://github.com/bloxbean/cardano-client-examples/blob/main/src/main/java/com/bloxbean/cardano/client/examples/MintAndBurn.java

It worked! https://testnet.cardanoscan.io/token/4d13f391e05ab6b76a3153fe861d8339573c2693ad0edf2f92b1ea17524d4232, Thank you very much.
